### PR TITLE
storage: TLS prepared statements

### PR DESCRIPTION
**Pull request by thibault-martinez**
- Extracts common sqlite3 wrappers in their own compilation unit
- Adds TLS prepared statements

# Test Plan:
CI

### CODE REVIEW

File: common/storage/connection.h [hunk 1]
```diff
@@ -18,13 +18,14 @@ extern "C" {
 
 typedef struct connection_t {
   void* db;
+  void* data;
```
Comment (th0br0) [inline common/storage/connection.h]: Can't we make connection opaque?

**Review by th0br0 — CHANGES_REQUESTED** [review]

File: common/storage/connection.h (hunk 1 above)
Comment (thibault-martinez) [reply common/storage/connection.h]: Will do

File: common/storage/sql/sqlite3/connection.c [hunk 2]
```diff
@@ -5,56 +5,229 @@
...
+                           iota_statement_milestone_exist_by_hash);
+  ret |= prepare_statement(connection->db,
+                           &connection->statements.state_delta_store,
+                           iota_statement_state_delta_store);
+  ret |= prepare_statement(connection->db,
+                           &connection->statements.state_delta_load,
+                           iota_statement_state_delta_load);
+
+  if (ret != RC_OK) {
+    log_error(SQLITE3_LOGGER_ID, "Preparing statements failed\n");
```
Comment (th0br0) [inline common/storage/sql/sqlite3/connection.c:117]: Let's also log the actual error here via `RC_RESOLVE_FORMAT_STR`

File: common/storage/sql/sqlite3/connection.c [hunk 3]
```diff
@@ -5,56 +5,229 @@
...
+  ret |= finalize_statement(connection->statements.milestone_select_first);
+  ret |= finalize_statement(connection->statements.milestone_select_last);
+  ret |= finalize_statement(connection->statements.milestone_select_next);
+  ret |= finalize_statement(connection->statements.milestone_exist);
+  ret |= finalize_statement(connection->statements.milestone_exist_by_hash);
+  ret |= finalize_statement(connection->statements.state_delta_store);
+  ret |= finalize_statement(connection->statements.state_delta_load);
+
+  if (ret != RC_OK) {
+    log_error(SQLITE3_LOGGER_ID, "Finalizing statements failed\n");
```
Comment (th0br0) [inline common/storage/sql/sqlite3/connection.c:169]: Let's also log the actual error here via `RC_RESOLVE_FORMAT_STR`

File: common/storage/connection.h [hunk 4]
```diff
@@ -8,25 +8,23 @@
...
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-typedef struct connection_t {
-  void* db;
-} connection_t;
+typedef struct opaque_connection_s {
+  void* actual;
+} opaque_connection_t;
```
Comment (th0br0) [inline common/storage/connection.h]: let's call this `storage_connection_t`?

**Review by th0br0 — APPROVED** [review]

**Pull request merged**